### PR TITLE
[fix] composite compliance: quantile and nanquantile

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference https://github.com/pytorch/pytorch/issues/69991

Refactored such that only `out` variant copies the result into `out` otherwise we just return the result of the composite functions as is.

### CODE REVIEW

**Comment by kshitij12345** [comment]
@zou3519 Gentle ping :)

File: aten/src/ATen/native/Sorting.cpp [hunk 1]
```diff
@@ -185,8 +179,19 @@ void quantile_impl(
...
+}
+
+Tensor quantile_compute(
+    const Tensor& self,
+    const Tensor& q,
+    const optional<int64_t> _dim,
+    const bool keepdim,
+    const QUANTILE_INTERPOLATION_MODE& interpolation,
+    const bool ignore_nan,
+    int64_t dim,
```
Comment (zou3519) [inline aten/src/ATen/native/Sorting.cpp:193]: nit: we should probably name `_dim` and `dim` better. Something like `original_dim` and one of {`wrapped_dim`, `sanitized`, `inferred_dim`} ?

**Review by zou3519 — APPROVED** [review]
LGTM. I had a minor nit, let me know if you want to resolve that before I land this

File: aten/src/ATen/native/Sorting.cpp (hunk 1 above)
Comment (kshitij12345) [reply aten/src/ATen/native/Sorting.cpp:193]: Yeah. That makes sense. Will do that in sometime and ping you. Thanks :)

Comment (kshitij12345) [reply aten/src/ATen/native/Sorting.cpp]: Done

**Pull request closed without merging**